wordpress-mobile/WordPress-Android
# Per app language preferences

This PR adds Per-app Language Preferences introduced in Android 12 (API 31) level.

Fixes #

App Lanaguages > App

https://user-images.githubusercontent.com/990349/190540086-ec27306b-7149-4621-92af-9923061c6a39.mp4

App > Language

https://user-images.githubusercontent.com/990349/190540143-37ae77a9-a96c-4552-944f-2b02b8626981.mp4

To test:

Test 1:
- Build and run
- Launch WordPress or Jetpack app
- Now go to device **Settings > System > Languages & Input > App Languages** > Select (WordPress app or Jetpack app)
- Ensure WordPress or Jetpack app shows in App Lanaguages
- Tap on the app.  It should navigate to App Lanaguage selection screen

Test 2:
- Build and run
- Launch WordPress or Jetpack app
- **Settings > Apps > Select (WordPress app or Jetpack app) > Language** (scroll if required)
- Tap on the Language option.  It should navigate to App Lanaguage selection screen

NOTE: 
- WordPress app / Jetpack app already have in app Language picker ( Me > App Setttings > Interface Language)
- Once we upgrade **compileSdkVersion** and **targetSdkVersion** to Android 13 (API 33), we can enable this Per-app Language Preferences and potentially remove in app language picker


## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
